File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        List<Integer> l = new ArrayList<>();
        for (int i : lost) {
            l.add(i);
        }
        List<Integer> r = new ArrayList<>();
        for (int i : reserve) {
            r.add(i);
        }

        List<Integer> both = new ArrayList<>();
        for (int rr : r) {
            if (l.contains(rr)) {
                both.add(rr);
            }
        }

        r.removeAll(both);
        l.removeAll(both);

        Collections.sort(r);
        Collections.sort(l);

        for (int rr : r) {
            if (l.contains(rr - 1)) {
                l.remove(Integer.valueOf(rr - 1));
            } else if (l.contains(rr + 1)) {
                l.remove(Integer.valueOf(rr + 1));
            }
        }

        return n - l.size();
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    lost = set(lost)
    reserve = set(reserve)
    
    common = lost & reserve
    lost -= common
    reserve -= common
    
    for student in sorted(reserve):
        if student-1 in lost:
            lost.remove(student-1)
        elif student+1 in lost:
            lost.remove(student+1)
    return n-len(lost)